services/metadata_service/tests/integration_tests/utils.py

```diff
@@ -0,0 +1,251 @@
+import json
+import os
+
+import psycopg2
+import pytest
+from aiohttp import web
+from services.data.postgres_async_db import AsyncPostgresDB
+from services.utils import DBConfiguration
+from services.metadata_service.api.admin import AuthApi
+from services.metadata_service.api.flow import FlowApi
+from services.metadata_service.api.run import RunApi
+from services.metadata_service.api.step import StepApi
+from services.metadata_service.api.task import TaskApi
+from services.metadata_service.api.artifact import ArtificatsApi
+from services.metadata_service.api.metadata import MetadataApi
+
+# Migration imports
+from services.migration_service.api.admin import AdminApi as MigrationAdminApi
+from services.migration_service.data.postgres_async_db import \
+    AsyncPostgresDB as MigrationAsyncPostgresDB
+
+# Test fixture helpers begin
+
+
+def init_app(loop, aiohttp_client, queue_ttl=30):
+    app = web.Application()
+
+    # Migration routes as a subapp
+    migration_app = web.Application()
+    MigrationAdminApi(migration_app)
+    app.add_subapp("/migration/", migration_app)
+
+    FlowApi(app)
+    RunApi(app)
+    StepApi(app)
+    TaskApi(app)
+    AuthApi(app)
+    ArtificatsApi(app)
+    MetadataApi(app)
+
+    return loop.run_until_complete(aiohttp_client(app))
+
+
+async def init_db(cli):
+    db_conf = DBConfiguration()
+    # Make sure migration scripts are applied
+    migration_db = MigrationAsyncPostgresDB.get_instance()
+    await migration_db._init(db_conf)
+
+    # Apply migrations and make sure "is_up_to_date" == True
+    await cli.patch("/migration/upgrade")
+    # TODO: Api not responding with valid json headers so check fails.
+    # status = await (await cli.get("/migration/db_schema_status")).json()
+    # assert status["is_up_to_date"] is True
+
+    db = AsyncPostgresDB.get_instance()
+    await db._init(db_conf)
+    return db
+
+
+async def clean_db(db: AsyncPostgresDB):
+    # Tables to clean (order is important due to foreign keys)
+    tables = [
+        db.metadata_table_postgres,
+        db.artifact_table_postgres,
+        db.task_table_postgres,
+        db.step_table_postgres,
+        db.run_table_postgres,
+        db.flow_table_postgres
+    ]
+    for table in tables:
+        await table.execute_sql(select_sql="DELETE FROM {}".format(table.table_name))
```

Comment:
@ferras should we include some kind of guard for the test suite, so this can not be run against any non-test DB? currently there are no checks in place for the DSN provided by the DBConfiguration class, rather it is implicitly expected that when running tests, the environment is configured to point to a test DB.

this could have some serious effects in case a user ends up somehow running the test suite against a production db, specifically due to the `clean_db()` teardown.

Reply:
I propose that we fix the database hostname to `db_test` to avoid accidental test runs against wrong database, at least for now.

Reply:
yeah i think the db name (not just the hostname) and really the rest of the configuration (password, username) should be hard coded for the test to avoid any accidents

Reply:
I opened a PR with a more thorough solution for this configuration with #167 , which should effectively namespace used environment variables by the test suite. Can be merged into this branch if acceptable.